geo/src/algorithm/concave_hull.rs

```diff
@@ -0,0 +1,432 @@
+use crate::algorithm::convexhull::ConvexHull;
+use crate::algorithm::euclidean_distance::EuclideanDistance;
+use crate::algorithm::euclidean_length::EuclideanLength;
+use crate::prelude::Centroid;
+use crate::utils::partial_min;
+use crate::{Line, LineString, MultiLineString, MultiPoint, MultiPolygon, Point, Polygon};
+use num_traits::Float;
+use rstar::{RTree, RTreeNum};
+use std::collections::VecDeque;
+
+pub trait ConcaveHull<T> {
```

Comment:
Could you make `T` into an associated type `Scalar` like this:

``` rust
pub trait ConcaveHull {
    type Scalar: CoordinateType;
    fn concave_hull(...) -> Polygon<Self::Scalar>;
}
```

Generally, we want the trait to be generic only if the same type could implement the trait for two different `T` (eg. see `Iterator` vs `Add` traits in std).  I've generally been advocating for this change, and recently modified the `ConvexHull` trait to reflect that.